lib/reversetunnel/rc_manager.go

```diff
@@ -208,29 +208,33 @@ func (w *RemoteClusterTunnelManager) Sync(ctx context.Context) error {
 	return trace.NewAggregate(errs...)
 }
 
-func (w *RemoteClusterTunnelManager) realNewAgentPool(ctx context.Context, cluster, addr string) (*AgentPool, error) {
+func realNewAgentPool(ctx context.Context, cfg RemoteClusterTunnelManagerConfig, cluster, addr string) (*AgentPool, error) {
 	pool, err := NewAgentPool(ctx, AgentPoolConfig{
 		// Configs for our cluster.
-		Client:              w.cfg.AuthClient,
-		AccessPoint:         w.cfg.AccessPoint,
-		HostSigner:          w.cfg.HostSigner,
-		HostUUID:            w.cfg.HostUUID,
-		LocalCluster:        w.cfg.LocalCluster,
-		Clock:               w.cfg.Clock,
-		KubeDialAddr:        w.cfg.KubeDialAddr,
-		ReverseTunnelServer: w.cfg.ReverseTunnelServer,
-		FIPS:                w.cfg.FIPS,
+		Client:              cfg.AuthClient,
+		AccessPoint:         cfg.AccessPoint,
+		HostSigner:          cfg.HostSigner,
+		HostUUID:            cfg.HostUUID,
+		LocalCluster:        cfg.LocalCluster,
+		Clock:               cfg.Clock,
+		KubeDialAddr:        cfg.KubeDialAddr,
+		ReverseTunnelServer: cfg.ReverseTunnelServer,
+		FIPS:                cfg.FIPS,
 		// RemoteClusterManager only runs on proxies.
 		Component: teleport.ComponentProxy,
 
 		// Configs for remote cluster.
-		Cluster:   cluster,
-		ProxyAddr: addr,
+		Cluster:  cluster,
+		Resolver: StaticResolver(addr),
 	})
 	if err != nil {
 		return nil, trace.Wrap(err, "failed creating reverse tunnel pool for remote cluster %q at address %q: %v", cluster, addr, err)
 	}
-	go pool.Start()
+	go func() {
+		if err := pool.Start(); err != nil {
+			cfg.Log.WithError(err).Error("Failed to start agent pool")
+		}
+	}()
```

Comment:
Not sure why we were starting a background goroutine here, but it isn't necessary.  This can just be called inline.